### DISCUSSION TITLE
setting canvas font does not set font until after canvas is rendered - missing hook?

### DISCUSSION THREAD

**Discussion by cindyloo** [wojtekmaj/react-pdf#1670] [Q&A] [answered]
### Before you start - checklist

- [X] I followed instructions in documentation written for my React-PDF version
- [X] I have checked if this bug is not already reported
- [X] I have checked if an issue is not listed in [Known issues](https://github.com/wojtekmaj/react-pdf/wiki/Known-issues)
- [X] If I have a problem with PDF rendering, I checked if my PDF renders properly in [PDF.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)

### Description

my assumption is that getting the canvas context from `Page` and setting it means that I can alter the font ( looking at the example here: https://jsfiddle.net/cxrznbso/1/ ). However, it appears that any of my changes apply to *after* the first writing to the canvas. I realize I don't exactly know how the PDF *text* is rendered to the canvas, but it seems that inside `PageCanvas` there would be the options to pass in a Font spec inside `function drawPageOnCanvas()`?

Is there a way to override the default 'system' font in canvas after the loading of the PDF and before the content is first rendered? I've tried almost all the handlers that are exposed to `Page`

thank you

### Steps to reproduce

use a callback handler to try to change canvas font

```
   async function onLoadSuccess(page) {
    console.log("LoadSuccess", page);
    
      const cv = canvas.current as any;
      if (!!cv){
        const ctx = (cv as HTMLCanvasElement)?.getContext("2d");
          if (!!cv && !!ctx){
            ctx.font = "16px Courier";
        }
      }
   }


 <Page
        className="test"
        canvasRef={canvas}
          key={`page-${index}`}
          onLoadSuccess={onLoadSuccess}
          onRenderTextLayerSuccess={onRenderTextLayerSuccess}
          customTextRenderer={onCustomTextRendererSuccess}
          onGetTextSuccess={onGetTextSuccess}
          onRenderSuccess={onPageRenderSuccess}
          pageIndex={scrollToPage + index -1} // CSB needs to change so we can scroll back one page. not great
          width={containerWidth ? Math.min(containerWidth, maxWidth) : maxWidth}
  

        />
```

### Expected behavior

I would think that instead of the system font, I'd have the text in the desired Courier font:

### Actual behavior

<img width="128" alt="Screenshot 2023-11-21 at 12 16 46 PM" src="https://github.com/wojtekmaj/react-pdf/assets/659092/9b41e0d9-d5c7-4b9c-8e9e-310d02079c70">

### Environment

- **Browser (if applicable)**:Chrome
- **React-PDF version**: newest
- **React version**: 18
- **Webpack version (if applicable)**:

**Comment by cindyloo** [comment]
addendum, I also hid the canvas just to see if the textLayer would look ok with the new font. It has some skips and overwritings, so I don't think that is the right way to do it?

**Comment by wojtekmaj** [comment, marked as answer]
You can't alter PDF font. PDF.js will set a font for each node it renders. You can only set font for stuff you render yourself, like a watermark.

**Comment by cindyloo** [comment]
got it. I hid the canvas and wrote out the font that I wanted, but because it maps back to the original canvas/font, it just doesn't look quite right bc the glyph is always a little different.
